### PR TITLE
q2pro: init at r3510

### PR DESCRIPTION
**Pull request by carlossless**
## Description of changes

Enhanced Quake 2 client and server focused on multiplayer

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/q2/q2pro/package.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    libjpeg
+    curl
+    SDL2
+    libogg
+    libvorbis
+    ffmpeg
+  ] ++ lib.optionals stdenv.isLinux [
+    openal
+  ] ++ lib.optionals stdenv.isDarwin [
+    openalSoft
```
Comment (carlossless) [inline pkgs/by-name/q2/q2pro/package.nix:54]: [q2pro relies on <(Open)AL/atext.h>](https://github.com/skullernet/q2pro/blob/master/src/client/sound/qal.h#L29) which is not present in OpenAL.framework on macOS, therefore opting to use [openal-soft](https://github.com/kcat/openal-soft) instead.

Comment (carlossless) [reply pkgs/by-name/q2/q2pro/package.nix]: no longer relevant as the author [recommends using openal-soft for all cases]:(https://github.com/skullernet/q2pro/blob/r3510/INSTALL.md)
> OpenAL sound backend requires OpenAL Soft development headers for compilation. At runtime, OpenAL library from any vendor can be used (but OpenAL Soft is strongly recommended).

File: pkgs/by-name/q2/q2pro/package.nix [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  wayland-scanner,
+  makeWrapper,
+  versionCheckHook,
+  x11Support ? stdenv.isLinux,
+  waylandSupport ? stdenv.isLinux,
+}:
+
+stdenv.mkDerivation (finalAttrs: rec {
+  pname = "q2pro";
+  version = "r3510";
```
Comment (Aleksanaa) [inline pkgs/by-name/q2/q2pro/package.nix:30]: version should not prefix with a letter, see https://github.com/NixOS/nixpkgs/tree/master/pkgs#versioning

File: pkgs/by-name/q2/q2pro/package.nix [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+  src = fetchFromGitHub {
+    owner = "skullernet";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/q2/q2pro/package.nix:34]:
```suggestion
    repo = "q2pro";
```

File: pkgs/by-name/q2/q2pro/package.nix [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      makeWrapper $out/bin/q2pro-unwrapped $out/bin/q2pro \
+        --prefix ${ldLibraryPathEnvName} : "${lib.makeLibraryPath finalAttrs.buildInputs}"
+    '';
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  meta = {
+    description = "Enhanced Quake 2 client and server focused on multiplayer";
```
Comment (Aleksanaa) [inline pkgs/by-name/q2/q2pro/package.nix:102]: missing `homepage = "https://github.com/skullernet/q2pro";`

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**